CodePush.m

```diff
@@ -554,20 +523,65 @@ - (void)savePendingUpdate:(NSString *)packageHash
         NSError *error;
         NSDictionary *currentPackage = [CodePushPackage getCurrentPackage:&error];
         if (!error && currentPackage) {
-            NSString *currentPackageIdentifier = [self getPackageStatusReportIdentifier:currentPackage];
-            if (currentPackageIdentifier && [self isDeploymentStatusNotYetReported:currentPackageIdentifier]) {
-                [self recordDeploymentStatusReported:currentPackageIdentifier];
-                resolve(@{ @"package": currentPackage, @"status": DeploymentSucceeded });
-                return;
+            NSString *currentPackageIdentifier = [CodePushStatusReport getPackageStatusReportIdentifier:currentPackage];
+            NSString *previousStatusReportIdentifier = [CodePushStatusReport getPreviousStatusReportIdentifier];
+            if (currentPackageIdentifier) {
+                if (previousStatusReportIdentifier == nil) {
+                    [CodePushStatusReport recordDeploymentStatusReported:currentPackageIdentifier];
+                    resolve(@{
+                              @"package": currentPackage,
+                              @"status": DeploymentSucceeded
+                              });
+                    return;
+                } else if (![previousStatusReportIdentifier isEqualToString:currentPackageIdentifier]) {
```

Comment:
When would the current and previous update has the same identifier? Is that even possible?